// react-box-sizing/lib/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PaddingBox = exports.ContentBox = exports.BorderBox = undefined;

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _glamor = require('glamor');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var BoxSizing = function BoxSizing(_ref) {
  var children = _ref.children,
      value = _ref.value;

  return _react2.default.createElement(
    'div',
    (0, _glamor.css)({
      boxSizing: value,
      '& *': { boxSizing: value },
      '& *:after': { boxSizing: value },
      '& *:before': { boxSizing: value }
    }),
    children
  );
};

BoxSizing.propTypes = {
  children: _react2.default.PropTypes.node
};

var BorderBox = exports.BorderBox = function BorderBox(props) {
  return _react2.default.createElement(BoxSizing, _extends({ value: 'border-box' }, props));
};
var ContentBox = exports.ContentBox = function ContentBox(props) {
  return _react2.default.createElement(BoxSizing, _extends({ value: 'content-box' }, props));
};
var PaddingBox = exports.PaddingBox = function PaddingBox(props) {
  return _react2.default.createElement(BoxSizing, _extends({ value: 'padding-box' }, props));
};
